samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/event/ExportEventLogServlet.java

```diff
@@ -0,0 +1,108 @@
+package org.sakaiproject.tool.assessment.ui.servlet.event;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.List;
+import java.util.ResourceBundle;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.apache.commons.text.translate.CsvTranslators.CsvEscaper;
+import org.sakaiproject.component.cover.ComponentManager;
+import org.sakaiproject.samigo.util.SamigoConstants;
+import org.sakaiproject.time.api.UserTimeService;
+import org.sakaiproject.tool.assessment.data.dao.assessment.EventLogData;
+import org.sakaiproject.tool.assessment.services.assessment.EventLogService;
+
+public class ExportEventLogServlet extends HttpServlet {
+
+
+    private static final String CSV_SEPARATOR = ",";
+    private static final DateTimeFormatter EXPORT_DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+    private UserTimeService userTimeService = ComponentManager.get(UserTimeService.class);
+    private ResourceBundle resourceBundle = ResourceBundle.getBundle(SamigoConstants.EVENT_LOG_BUNDLE);
+
+    public static final String PARAM_SITE_ID = "siteId";
+    public static final String PARAM_ASSESSMENT_ID = "assessmentId";
+
+
+    public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
```

Comment:
Is this endpoint secured or everybody can request it? I don't see any protection, maybe it goes through the requestfilter? Are we checking permissions?

Reply:
Yes, It's not secures, I'll fix it!